muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/SyncStudentEventHandler.java

```diff
@@ -30,16 +37,30 @@
   @Inject
   private ChatSyncController chatSyncController;
   
+  @Inject
+  private ChatController chatController;
+  
   @Inject
   private WorkspaceEntityController workspaceEntityController;
   
+  @Inject
+  private WorkspaceUserEntityController workspaceUserEntityController;
+  
   public synchronized void onSchoolDataWorkspaceUserDiscoveredEvent(@Observes SchoolDataWorkspaceUserDiscoveredEvent event) {
     //kun kurssille pelmahtaa uusi opiskelija
-    
     String userIdentifier = event.getUserIdentifier();
     String userDataSource = event.getUserDataSource();
     SchoolDataIdentifier user = new SchoolDataIdentifier(userIdentifier, userDataSource);
-    chatSyncController.syncStudent(user);
+    String workspaceDataSource = event.getWorkspaceDataSource();
+    String workspaceIdentifier = event.getWorkspaceIdentifier();
+    
+    WorkspaceEntity workspace = workspaceEntityController.findWorkspaceByDataSourceAndIdentifier(workspaceDataSource, workspaceIdentifier);
+    
+    WorkspaceChatSettings workspaceChatStatus = chatController.findWorkspaceChatSettings(workspace.getId());
+
+    if (workspaceChatStatus.getStatus() == WorkspaceChatStatus.ENABLED) {
+      chatSyncController.syncStudent(user);
```

Comment:
This seems to sync the whole student (i.e. all the workspaces), it would probably be better if it only synchronized the student for the single workspace that is provided by the event. This works, but probably does a little extra work like this.